CDC-Data-Reconciliation-Backend/compare.py

```diff
@@ -104,11 +104,16 @@ def comp(state_dict, cdc_dict):
             
         else:
             i = 0
-            for attribute in state_row:
+            # Determine which attributes to compare: specified ones or all
+            attributes_to_compare = compare_attributes if compare_attributes is not None else state_row.keys()
+            for attribute in attributes_to_compare:
                 # Skip the first attribute since they are not included in CDC CSV file
                 if i < 1:
```

Comment:
We should just remove this way of skipping the first attribute (Lines 111-113) as this will cause the first attribute to be skipped if using compare_attributes. With the addition of the lines 114-116 it is no longer necessary.